Classes/Controller/NewTenantController.php

```diff
@@ -155,29 +227,22 @@ public function addMetadataAction()
      */
     public function addSolrCoreAction()
     {
-        $this->pid = (int) GeneralUtility::_GP('id');
-        // Build data array.
-        $data['tx_dlf_solrcores'][uniqid('NEW')] = [
-            'pid' => intval($this->pid),
-            'label' => $this->getLanguageService()->getLL('flexform.solrcore') . ' (PID ' . $this->pid . ')',
-            'index_name' => '',
-        ];
-        $_ids = Helper::processDBasAdmin($data);
-        // Check for failed inserts.
-        if (count($_ids) == 1) {
-            // Fine.
-            $this->addFlashMessage(
-                $this->getLanguageService()->getLL('flash.solrcoreAddedMsg'),
-                $this->getLanguageService()->getLL('flash.solrcoreAdded'),
-                \TYPO3\CMS\Core\Messaging\FlashMessage::OK
-            );
-        } else {
-            // Something went wrong.
-            $this->addFlashMessage(
-                $this->getLanguageService()->getLL('flash.solrcoreNotAddedMsg'),
-                $this->getLanguageService()->getLL('flash.solrcoreNotAdded'),
-                \TYPO3\CMS\Core\Messaging\FlashMessage::ERROR
-            );
+        $doPersist = false;
+
+        if ($this->solrCoreRepository->findOneByPid($this->pid) === null) {
+            $newRecord = GeneralUtility::makeInstance(SolrCore::class);
+            $newRecord->setLabel($this->getLanguageService()->getLL('flexform.solrcore') . ' (PID ' . $this->pid . ')');
+            $newRecord->setIndexName('');
```

Comment:
This didn't work for me. The database record was created, but no Solr core because now the DataHandler hook isn't called. Perhaps we could just create the core here and pass its name to `setIndexName()`?

Reply:
Good point. So my setup just recreated the deleted solr record. I'll think about it.

Reply:
I think, it's fixed now.
In the DataHandler we need this part, too because you can just create the Solr core record via the List module.